api/src/client/dynamics/license-status/RgbDynamicsLicenseApplicationIdsClient.test.ts

```diff
@@ -0,0 +1,449 @@
+/* eslint-disable unicorn/no-null */
+import {
+  appTypeKeysString,
+  licenseAppType,
+  RgbDynamicsLicenseApplicationIdsClient,
+} from "@client/dynamics/license-status/RgbDynamicsLicenseApplicationIdsClient";
+import {
+  BusinessIdAndName,
+  LicenseApplicationIdsForAllBusinessIdsClient,
+} from "@client/dynamics/license-status/rgbDynamicsLicenseStatusTypes";
+import { NO_MATCH_ERROR } from "@domain/types";
+import { LogWriter, LogWriterType } from "@libs/logWriter";
+import { generateLicenseApplicationIdApiResponseValue } from "@test/factories";
+import {
+  HEALTH_CLUB_LICENSE_UUID,
+  PUBLIC_MOVERS_LICENSE_UUID,
+  RGB_APP_TYPE_KEYS,
+  RGB_LICENSE_APPLICATION_INFORMATION,
+} from "@test/helpers";
+import axios from "axios";
+
+jest.mock("axios");
+jest.mock("winston");
+const mockAxios = axios as jest.Mocked<typeof axios>;
+
+describe("RgbDynamicsLicenseApplicationIdClient", () => {
+  let client: LicenseApplicationIdsForAllBusinessIdsClient;
+  let logger: LogWriterType;
+
+  const ORG_URL = "www.test-org-url.com";
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    logger = LogWriter("NavigatorWebService", "ApiLogs", "us-test-1");
+    client = RgbDynamicsLicenseApplicationIdsClient(logger, ORG_URL);
+  });
+
+  const mockBusinessNameAndId_1: BusinessIdAndName = {
+    name: "Example Business",
+    businessId: "1234567890",
+  };
+  const mockBusinessNameAndId_2: BusinessIdAndName = {
+    name: "Example Business",
+    businessId: "0987654321",
+  };
+  const mockAccessToken = "access-granted";
+
+  describe("core functionality", () => {
+    it("makes a single get request to dynamics api with the license application id query and auth token in header", async () => {
+      const mockLicenseApplicationIdResponse = {
+        value: [
+          generateLicenseApplicationIdApiResponseValue({
+            rgb_name: "Example Business-Public Movers",
+            rgb_number: "123400",
+          }),
+        ],
+      };
+
+      mockAxios.get.mockResolvedValue({ data: mockLicenseApplicationIdResponse });
+      await client.getLicenseApplicationIdsForAllBusinessIds(mockAccessToken, [mockBusinessNameAndId_1]);
+      expect(mockAxios.get).toHaveBeenCalledWith(
+        `${ORG_URL}/api/data/v9.2/rgb_applications?$select=${appTypeKeysString},_rgb_apptypeid_value,rgb_name,rgb_appnumber,rgb_number,rgb_startdate,rgb_versioncode,rgb_expirationdate,statecode,statuscode,rgb_applicationid&$filter=(_rgb_businessid_value eq ${mockBusinessNameAndId_1.businessId})`,
```

Comment:
[dust] You can make the URL a const at the top of the file and append the part you are expecting to change if you want to make it more readable